### PR TITLE
IAV-1825 Health Questionnaire (slider update)

### PR DESCRIPTION
**Pull request by juhog**
Summary:
- jQuery UI better css file (with only structure, no theme)
- Base style 'visually-hidden'
- Slider update (js, scss, twig)

I suggest taking a look at the commits one by one to see what this PR contains.

I'm not sure if jquery-ui-touch-punch library should be added to the styleguide to fix mobile sliders. Currently it's not part of this PR.

----

Screenshot from local. Github public version of the slider won't work until slider.js related problem is solved (mentioned in Slack).

screen shot 2016-12-14 at 23 27 09

### CODE REVIEW

**Comment by lauriii** [comment]
I think the library to add this functionality to mobile should be added as part of this PR.

**Comment by juhog** [comment]
> I think the library to add this functionality to mobile should be added as part of this PR.

@lauriii This is now fixed. I added jQuery Touch Punch to the styleguide in this commit: https://github.com/LaakarikeskusAava/styleguide/pull/12/commits/81e268da5d6b091f2a167d1a213a5030a66ba282

There is still one problem in the slider though. The slider works in local but it doesn't work in the public github page. See this slack message: https://druid.slack.com/archives/aava-web/p1481729948000210

**Comment by juhog** [comment]
> There is still one problem in the slider though. The slider works in local but it doesn't work in the public github page. See this slack message: https://druid.slack.com/archives/aava-web/p1481729948000210

This problem should be fixed is now. Component .js files are copied to public directory. See this commit https://github.com/LaakarikeskusAava/styleguide/pull/12/commits/fd8196b796d2c8ff7d1bde13b8bddd56a2f96ca9 and the screens below.

screen shot 2017-03-25 at 09 56 11
screen shot 2017-03-25 at 09 57 07

---

@lauriii Is this PR ok to merge now?

**Comment by juhog** [comment]
@lauriii Can I get a review for this PR? (There's one fix that's gonna help the new form style PR too.)

File: styles/components/forms/slider/slider.json [hunk 1]
```diff
@@ -1,9 +1,5 @@
...
-  "elements": [
-    {
-      "name": "demo",
-      "min": 0,
-      "max": 10
-    }
-  ]
+  "min": 0,
+  "max": 10,
+  "input": "<input type=\"range\" min=\"0\" max=\"10\" value=\"5\">"
```
Comment (lauriii) [inline styles/components/forms/slider/slider.json:4]: Why do we have html in the json?

Comment (juhog) [reply styles/components/forms/slider/slider.json:4]: Front-end details in Finnish.

En nyt tarkkaan muista ilman että kaivelis vanhoja tietoja.

Muistaakseni kuitenkin idea on se, että slider-komponentille annetaan form-elementti, ja slider-komponentti muuttaa sen sitten tollaseks jquery ui slideriksi. Kun slideria käyttää niin sen form-elementin arvo muuttuu. Form-elementti voi olla `type="range"` tai `type="text"` tai vaikka `<select>`.

Comment (juhog) [reply styles/components/forms/slider/slider.json:4]: @lauriii Does the comment above seem valid? I think this is our only component that has a .json file though.

**Comment by juhog** [comment]
(Test comment 1 - Does this show up in Slack?)

File: styles/components/forms/slider/slider.json (hunk 1 above)
Comment (juhog) [reply styles/components/forms/slider/slider.json:4]: @lauriii Ping. See comments above.

Comment (lauriii) [reply styles/components/forms/slider/slider.json:4]: This seems fine, even though I still think its weird

**Comment by juhog** [comment]
(Test comment 2 - Does this show up in Slack?)

**Pull request merged**